lib/proxy/util.rb

```diff
@@ -27,8 +28,13 @@ def start(&ensured_block)
           logger.info "[#{thr.pid}] Started task #{cmdline_string}"
           stdin.write(input) if input
           stdin.close
-          stdout.each do |line|
-            logger.debug "[#{thr.pid}] #{line}"
+          unless @output.nil?
+            File.binwrite(@output, stdout.read)
+            stdout.close
+          else
+            stdout.each do |line|
+              logger.debug "[#{thr.pid}] #{line}"
+            end
```

Comment:
I think the debug logging behavior should remain, regardless of the presence of `@output`. If not, at least the comments above the method should reflect the implementation.

It may be better to introduce a new method to "pipe" data.

Reply:
What output would you like to print here? Even if we pipe data into the file, there won't be any kind of special output since we want all of it to be written to the file. Moreover, it's a binary file so I wouldn't recommend attaching this to some debug log.